cmd/scollector/collectors/nexpose.go

```diff
@@ -0,0 +1,405 @@
+	EndTime   string `xml:"endTime,attr"`
+	Status    string `xml:"status,attr"`
+}
+
+func init() {
+	registerInit(func(c *conf.Conf) {
+		for _, n := range c.Nexpose {
+			collectors = append(collectors, &IntervalCollector{
+				F: func() (opentsdb.MultiDataPoint, error) {
+					return c_nexpose(n.Username, n.Password, n.Host, n.Insecure)
+				},
+				name: fmt.Sprintf("nexpose-%s", n.Host),
+			})
+		}
+	})
+}
+
+type nexposeConnection struct {
+	Username  string
+	Password  string
+	Host      string
+	URLv1     string
+	URLv2     string
+	Client    *http.Client
+	SessionID string
+	Cookie    *http.Cookie
+}
+
+func c_nexpose(username, password, host string, insecure bool) (opentsdb.MultiDataPoint, error) {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: insecure},
+	}
+	client := &http.Client{Transport: tr}
+	url1 := fmt.Sprintf("https://%s/api/1.1/xml", host)
+	url2 := fmt.Sprintf("https://%s/api/1.2/xml", host)
+
+	var md opentsdb.MultiDataPoint
+
+	c := nexposeConnection{Username: username, Password: password, URLv1: url1, URLv2: url2, Host: host, Client: client}
+
+	err := c.login()
+	if err != nil {
+		return nil, fmt.Errorf("Login failed: %s", err)
+	}
+
+	siteSummaries, err := c.siteListing()
+	if err != nil {
+		return nil, err
+	}
+
+	var siteNames = map[int]string{}
+
+	for _, site := range *siteSummaries {
+		siteNames[site.ID] = site.Name
+		Add(&md, "nexpose.site.risk_score", site.RiskScore, opentsdb.TagSet{"site": site.Name}, metadata.Gauge, metadata.Score, descRiskScore)
+		tags := opentsdb.TagSet{"site": site.Name}
+
+		assets, err := c.getSiteAssets(site.ID)
+		if err != nil {
+			return nil, err
+		}
+		for _, asset := range *assets {
+			now := time.Now()
+			last_scan := now.Unix() - (asset.LastScanDate / 1000)
+			var assetTags opentsdb.TagSet
+			// Set the AssetName to the hostname if we have it, otherwise the IP.
+			if asset.AssetName != "" {
+				assetTags = opentsdb.TagSet{"name": asset.shortName()}.Merge(tags)
+			} else {
+				assetTags = opentsdb.TagSet{"name": asset.AssetIP}.Merge(tags)
+			}
+			Add(&md, "nexpose.asset.risk_score", asset.RiskScore, assetTags, metadata.Gauge, metadata.Score, descRiskScore)
+			Add(&md, "nexpose.asset.vuln_count", asset.VulnCount, assetTags, metadata.Gauge, metadata.Vulnerabilities, descVulnCount)
+			Add(&md, "nexpose.asset.exploit_count", asset.ExploitCount, assetTags, metadata.Gauge, metadata.Vulnerabilities, descExploitCount)
+			Add(&md, "nexpose.asset.malware_count", asset.MalwareCount, assetTags, metadata.Gauge, metadata.Vulnerabilities, descMalwareCount)
+			Add(&md, "nexpose.asset.last_scan", last_scan, assetTags, metadata.Gauge, metadata.Second, descAssetLastScan)
+
+			site.VulnCount += asset.VulnCount
+			site.ExploitCount += asset.ExploitCount
+			site.MalwareCount += asset.MalwareCount
+		}
+
+		Add(&md, "nexpose.site.vuln_count", site.VulnCount, tags, metadata.Gauge, metadata.Vulnerabilities, descVulnCount)
```

Comment:
Probably not. There are a huge number of devices for some sites and the risk scores have a humongous range, so average loses meaning pretty quickly. Median might be worth considering.